yt_dlp/extractor/vrsquare.py

```diff
@@ -0,0 +1,191 @@
+    _TESTS = [{
+        'url': 'https://livr.jp/contents/P470896661',
+        'info_dict': {
+            'id': 'P470896661',
+            'ext': 'mp4',
+            'title': 'そこ曲がったら、櫻坂？ ７年間お疲れ様！菅井友香の卒業を祝う会！前半 2022年11月6日放送分',
+            'description': 'md5:523726dc835aa8014dfe1e2b38d36cd1',
+            'duration': 1515.0,
+            'tags': list,
+            'thumbnail': 're:^https?://media.livr.jp/vod/img/.*$',
+        },
+    }, {
+        'url': 'https://livr.jp/contents/P589523973',
+        'info_dict': {
+            'id': 'P589523973',
+            'ext': 'mp4',
+            'title': '薄闇に仰ぐ しだれ桜の妖艶',
+            'description': 'md5:a042f517b2cbb4ed6746707afec4d306',
+            'duration': 1084.0,
+            'tags': list,
+            'thumbnail': 're:^https?://media.livr.jp/vod/img/.*$',
+        },
+        'skip': 'Paid video',
+    }, {
+        'url': 'https://livr.jp/contents/P316939908',
+        'info_dict': {
+            'id': 'P316939908',
+            'ext': 'mp4',
+            'title': '2024年5月16日（木） 「今日は誰に恋をする？」公演 小栗有以 生誕祭',
+            'description': 'md5:2110bdcf947f28bd7d06ec420e51b619',
+            'duration': 8559.0,
+            'tags': list,
+            'thumbnail': 're:^https?://media.livr.jp/vod/img/.*$',
+        },
+        'skip': 'Premium channel subscribers only',
+    }, {
+        # Accessible only in the VR SQUARE app
+        'url': 'https://livr.jp/contents/P126481458',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        status = self._download_json(
+            f'{self._BASE_URL}/webApi/contentsStatus/{video_id}',
+            video_id, 'Checking contents status', fatal=False)
+        if traverse_obj(status, 'result_code') == '40407':
+            self.raise_login_required('Unable to access this video')
+
+        try:
+            web_api = self._download_json(
+                f'{self._BASE_URL}/webApi/play/url/{video_id}', video_id)
+        except ExtractorError as e:
+            if isinstance(e.cause, HTTPError) and e.cause.status == 500:
+                raise ExtractorError('VR SQUARE app-only videos are not supported', expected=True)
+
+        return {
+            'id': video_id,
+            'title': self._html_search_meta(['og:title', 'twitter:title'], webpage),
+            'description': self._html_search_meta('description', webpage),
+            'formats': self._extract_m3u8_formats(traverse_obj(web_api, (
+                'urls', ..., 'url', any)), video_id, 'mp4', fatal=False),
+            'thumbnail': self._html_search_meta('og:image', webpage),
+            **traverse_obj(webpage, {
+                'duration': ({find_element(cls='layout-product-data-time')}, {parse_duration}),
+                'tags': ({find_elements(cls='search-tag')}, ..., {clean_html}),
+            }),
+        }
+
+
+class VrSquarePlaylistBaseIE(VrSquareIE):
```

Comment:
You should not inherit from a concrete IE
```suggestion
class VrSquarePlaylistBaseIE(InfoExtractor):
    _BASE_URL = 'https://livr.jp'
```